### PR TITLE
v3: Restrict Flagged CNR installs on non-local listeners

### PR DESCRIPTION
**Pull request by ltdrdata**
## Purpose and scope

A successful Registry install response previously allowed a Flagged CNR version to proceed. Require Flagged versions to satisfy the same installation conditions as Active versions, plus a server listening exclusively on loopback or an explicit administrator override. Covers installation, reinstallation, switching, updates, deferred startup and server-scheduled restores.

The allow_flagged_nodepack_install option belongs in config.ini’s [default] section. It defaults to false; only case-insensitive true enables it, after restarting ComfyUI. The override is intended for a trusted private network and permits non-loopback listeners. It does not detect private networks or bypass common security checks, Registry refusals or Git/pip restrictions. Active and Pending installation behavior is retained.

## Key changes

- Use status from the existing CNR install response. Check policy before changing the current pack and reuse that response for execution, without another Registry lookup.
- Validate reinstallation before removal, then install and rerun the pack’s script. Preserve files and activation state on denial, retain the reason in queue/update results and avoid retrying rejected restore targets.
- Recheck deferred work against current startup settings and pass the resolved server permission to scheduled restore children. Standalone CLI retains its existing interface and local administrator behavior.
- Show a short security-configuration denial in the UI. Only terminal logs provide the loopback listener and private-network override remedies, including restart instructions.

Affected modules: CNR metadata mapping, installation core, server queues, startup processing, configuration and README. Deferred switches use stored status without refreshing it from the Registry. Older reservations without status require loopback-only listeners or the override; otherwise startup rejects them without changing the pack and tells the user to request installation again. Enabling an already installed identical version remains a local activation with no Registry lookup or installation script.

## Validation approach

Core tests run production installation paths with controlled external responses. Server E2E uses real HTTP requests, queues, ZIP installation, restart and CLI subprocesses, with a local Registry and isolated paths, settings and databases. Live Chromium scenarios also exercise the combined UI through real routes and WebSocket completion.

| Intent | Observed result |
|---|---|
| Add restrictions only to Flagged | Active/Pending controls install; Flagged needs loopback or override. Strong-security controls reject both Active and Flagged before metadata lookup. |
| Preserve state on denial | Existing files, versions and activation state survive; no download or reservation is created. |
| Avoid duplicate lookups | Count metadata requests through installation and restore; deferred switches make no additional lookup. |
| Preserve execution context | Check immediate/deferred execution, old reservations, callback timing, standalone CLI behavior and actual server-scheduled restore children. Existing-version activation requires no Registry lookup or installation script. |
| Keep errors and settings correct | Verify short UI errors versus terminal remedies, Registry refusal, listener edge cases and preservation of an administrator's option during an unrelated UI save. |

Tests fail when status propagation or reinstall script execution is deliberately removed, and pass when restored. Allowed installation controls also pass.

## Validation results

| Scope | Result |
|---|---|
| Standalone 1796c72f, non-E2E regression | 89 passed; 22 subtests passed |
| UI + Flagged integration cc3d1001, non-E2E regression | 235 passed; 221 subtests passed |
| Policy server E2E on the final production files | 60 passed; 6 skipped |
| Live browser E2E on the final production files | 7 scenarios passed; no uncaught browser exceptions |

The six policy E2E skips require a preinstalled pack under strong security. Strong-security denial is tested directly. The full policy suite passed when run alone after an initial parallel run encountered a test-server port collision. No product code or assertions were changed to address that fixture collision.

Snapshot checks establish admission, extraction and denial-state preservation. Fresh-snapshot install.py completion and file-lock/deferred-deletion handling remain outside this PR.

Fixture E2E uses ComfyUI 039ed38e on Python 3.10.20 and frontend 1.45.15; other tests use Python 3.12. v3 starts without --enable-manager. Browser checks use single-tab sessions. The public Registry was not exercised in this rerun.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -458,6 +458,50 @@ Changes take effect after a **restart** (no hot reload).
...
+### Flagged CNR versions: `allow_flagged_nodepack_install`
+
+Manager checks the version status in the existing CNR install response; no
+additional Registry request is needed. `NodeVersionStatusFlagged` versions
+can be installed only when **all** `--listen` addresses are loopback, such as
+`127.0.0.1` or `::1`. A mixed loopback/non-loopback listener is non-local.
+Active versions continue to install normally, subject to the existing security
+policy. Registry refusals (including HTTP 404) still prevent installation.
+Flagged versions must satisfy those same conditions, with the additional
+loopback-only listener or explicit override requirement below.
```
[the remark this answers is not in the document]
Comment (ltdrdata) [reply README.md]: Accepted and clarified in 1796c72f. The Flagged rule now states the loopback requirement and explicit override together. Active behavior and the common Registry/security checks are described separately, without the ambiguous reference. This is a README-only correction.
